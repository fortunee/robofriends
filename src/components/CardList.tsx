import React from 'react';
import { RobotProps } from '../containers/App';
import Card from './Card';

interface CardListProps {
  robots: Array<RobotProps>
}

const CardList = ({ robots }: CardListProps) => (
  <>
    {robots.map((robot) => (
      <Card
        key={robot.id}
        id={robot.id}
        name={robot.name}
        email={robot.email}
      />
    ))}
  </>
);

export default CardList;
