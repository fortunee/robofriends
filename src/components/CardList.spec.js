import { shallow } from 'enzyme';
import React from 'react';

import CardList from './CardList';

it('Should render a CardList component', () => {
  const mockData = [
    {
      id: 1,
      name: 'Jane Doe',
      username: 'jdoe',
      email: 'jane@doe.random'
    }
  ]
  expect(shallow(<CardList robots={mockData} />).debug()).toMatchSnapshot();
});
