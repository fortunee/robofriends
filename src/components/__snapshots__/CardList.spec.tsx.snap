// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a CardList component 1`] = `
"<Fragment>
  <Card id={1} name=\\"Jane Doe\\" email=\\"jane@doe.random\\" />
</Fragment>"
`;
